### PR TITLE
[WIP] Update Packer Plugin documentation

### PR DESCRIPTION
**Pull request by nywilken**
- [ ] Add a FAQ page with some of the basic Extending Packer questions around v1.7.0, plugin breakouts, and the Packer plugin SDK.
- [ ] Add Packer Partner Development Program documentation

### CODE REVIEW

File: website/content/docs/plugins/creation/plugin-faqs.mdx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  1. Releasing new versions of the plugin binary that are compatible with Packer - See [Creating a Release](https://www.packer.io/docs/plugins/creation#creating-a-github-release) for more details.
+
+2. **What happens to an active plugin that is no longer being maintained?**
+Plugins that are not actively maintained will be marked as unmaintained on their respective documentation pages, and scheduled for archival. We will try to find a new owner. If no owner can be found then HashiCorp will follow the process for archiving a plugin that has no owner.
+
+3. **What does it mean for a plugin to be archived?**
+A plugin that does not have an owner will be broken out into its own repository under the HashiCorp GitHub organization. The new repository will then be [archived](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/archiving-a-github-repository) to let users know that the plugin is no longer being maintained. Upon the initial breakout, a final compatible release of the plugin will be made, but it will not be actively maintained or monitored for new issues. Critical security-related issues will be reviewed and addressed where possible.
```
Comment (SwampDragons) [inline website/content/docs/plugins/creation/plugin-faqs.mdx:20]: this "breakout" has ultimately already happened, so feels confusing to list here.

File: website/content/docs/plugins/creation/plugin-faqs.mdx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+4. **I've upgraded my plugin to use the latest Packer plugin SDK will it work versions of Packer prior to v1.7.0**?
+In Packer v1.7.0, a new Packer Plugin SDK was made available to assist maintainers in the creation of Packer plugins. In this new SDK a number of breaking changes were introduced so the plugin API version was bumped from version 4 to version 5.0. Thus making any plugin using the latest SDK incompatible with versions of Packer before v1.7.0.
+The latest API version supports both a major and a minor version to allow for minor improvements to the Packer Plugin SDK API without causing backwards incompatibilities for plugins on the same major API version.
+
+5. **Will existing plugins not using the new Packer Plugin SDK continue to work with Packer v1.7.0+**?
+Packer plugins built against version 4 of the Packer plugin API, those plugins not using github.com/hashicorp/packer-plugin-sdk, will continue to work with the latest versions of Packer but they must be installed manually. Support for plugin installation via the `packer init` command is only supported for plugins using the latest Packer Plugin SDK.
```
Comment (SwampDragons) [inline website/content/docs/plugins/creation/plugin-faqs.mdx:27]: wait, version 4 or 5? I thought version 4 doesnt work with the current core

**Review by SwampDragons — COMMENTED** [review]
Awesome start. A couple of questions inline.

File: website/content/docs/plugins/creation/plugin-faqs.mdx (hunk 2 above)
Comment (sylviamoss) [reply website/content/docs/plugins/creation/plugin-faqs.mdx:27]: +1 on Megan's comment. Here's the table that always help me with that https://www.packer.io/guides/1.7-template-upgrade#what-if-the-plugin-you-need-is-not-upgraded-to-api-v5-0

**Pull request closed without merging**